### ISSUE TITLE
UserSecretsId causing Microsoft.Extensions.DependencyInjection.Abstractions: Unable to resolve service for type X while attempting to activate Function.Y

### ISSUE THREAD

**Issue by jwadaptivity** [Azure/azure-functions-dotnet-extensions#45]
After publishing to Azure via VS2019 user secrets is populated as below 
```
<PropertyGroup>
	<TargetFramework>netcoreapp3.1</TargetFramework>
	<AzureFunctionsVersion>v3</AzureFunctionsVersion>
	<UserSecretsId>f360954f....</UserSecretsId>
</PropertyGroup> 
```

Which is causing 
Microsoft.Extensions.DependencyInjection.Abstractions: Unable to resolve service for type 'Interface' while attempting to activate 'Function'
Only seems to occur for services injected into dependent class libraries (Common) in this case. Removing UserSecretsId and it works fine.

Full broken .csproj
```
<Project Sdk="Microsoft.NET.Sdk">
	<PropertyGroup>
		<TargetFramework>netcoreapp3.1</TargetFramework>
		<AzureFunctionsVersion>v3</AzureFunctionsVersion>
		<UserSecretsId>f360954f... etc</UserSecretsId>
	</PropertyGroup>
	<ItemGroup>
		<PackageReference Include="FluentValidation" Version="9.0.0" />
		<PackageReference Include="Microsoft.Azure.Functions.Extensions" Version="1.0.0" />
		<PackageReference Include="Microsoft.NET.Sdk.Functions" Version="3.0.3" /> 
	</ItemGroup>
	<ItemGroup>
		<ProjectReference Include="..\Common\Common.csproj" />
	</ItemGroup>
	<ItemGroup>
		<None Update="host.json">
			<CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
		</None>
		<None Update="local.settings.json">
			<CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
			<CopyToPublishDirectory>Never</CopyToPublishDirectory>
		</None>
	</ItemGroup>
</Project>

```

**Comment by zhpeih** [comment]
I have encountered a similar problem without any errors, but the startup function is not executed. Removing UserSecretsId and it works fine.

**Comment by DabNote** [comment]
I had the same problem, after publishing my project to a Azure Function App a UserSecretsId was added to my project and my dependency injection stopped working.

Removing UserSecretsId and it started to work fine. But this wasn't a solution I was happy with.

I finally found I need to install Microsoft.Extensions.Configuration.UserSecrets and dependency injection started to work again. Makes sense now, was pulling my hair out. Hope this helps someone.
